docs/learn/prompts.md

```diff
@@ -1,219 +1,425 @@
+    genre: str
+    age_group: str
+
+
+prompt = BookRecommendationPrompt(genre="fantasy", age_group="adult")
+print(prompt.dump()['metadata'])
+# > {'tags': {'version:0001', 'category:books'}}
 ```
 
-You should note that this is just one example, and you can use all Pydantic functionalities to improve your prompts. Check out their docs here[link].
+!!! note "Adding Additional Fields"
+
+    Although `Metadata` is a `TypedDict` with only the `tags` key, there is nothing stopping you from adding additional keys. The only issue is that this will throw a type error, which you can ignore. If there are particular keys you find yourself using frequently, let us know so we can add it!
 
-## Tags
+### Running Prompts
 
-You may have noticed that the `dump()` method contains the key `tags`. Use the `tags` decorator to tag your prompts, useful for organization when logging your results.
+One of the key benefits of `BasePrompt` is that it is provider-agnostic. You can use the same prompt with different LLM providers, making it easy to compare performance or switch providers.
+
+You can do this with the `run` and `run_async` methods that run the prompt using the configuration of a call decorator, which  For now, just know that the decorator is configuring a call to the LLM, and the return value of the `run` and `run_async` methods match that of the decorator:
 
 ```python
-from mirascope.core import tags
+from mirascope.core import BasePrompt, anthropic, openai, prompt_template
+
 
-@tags(["version:0001"])
+@prompt_template("Recommend a {genre} book")
 class BookRecommendationPrompt(BasePrompt):
-    """Recommend a {genre} book."""
-		
     genre: str
-  
+
+
 prompt = BookRecommendationPrompt(genre="fantasy")
-print(prompt.dump())
-# > {
-#     'tags': ['version:0001'],
-#     ...
-#   }
+
+# Running the prompt with OpenAI
+print(prompt.run(openai.call(model="gpt-4o-mini")))
+# > Sure! If you're looking for a captivating fantasy novel, I recommend...
+
+# Running the prompt with Anthropic
+print(prompt.run(anthropic.call(model="claude-3-5-sonnet-20240620")))
+# > There are many great fantasy books to choose from, but...
 ```
+
+We will begin covering these decorators in more detail in the [following section](./calls.md).
+
+!!! note "Agnostic Assuming Support"
+
+    While `BasePrompt` is provider-agnostic, some features (like multi-modal inputs) may not be supported by all providers. We try to maximize support across providers, but you should always check the provider's capabilities when using more advanced features.
+
+## Best Practices and Use Cases
+
+- Provider Comparison: Use `BasePrompt` to easily test the same prompt across different providers to compare performance and output quality.
+- Prompt Versioning: Utilize the metadata decorator to keep track of different versions of your prompts as you refine them.
+- Dynamic Content: Leverage `@computed_field` for injecting dynamic content or API calls into your prompts.
```

Comment:
should we mention cached_property as well?

Reply:
Done.